istioctl/cmd/sidecar.go

```diff
@@ -0,0 +1,291 @@
+					"metadata": map[string]interface{}{
+						"name":      name,
+						"namespace": namespace,
+					},
+				},
+			}
+			spec := &networkingv1alpha3.WorkloadGroup{
+				Labels:         convertToStringMap(labels),
+				Ports:          convertToUnsignedInt32Map(ports),
+				Network:        network,
+				ServiceAccount: serviceAccount,
+			}
+			wgYAML, err := generateWorkloadGroupYAML(u, spec)
+			if err != nil {
+				return err
+			}
+			_, err = cmd.OutOrStdout().Write([]byte(wgYAML))
+			return err
+		},
+	}
+	createGroupCmd.PersistentFlags().StringVar(&name, "name", "", "The name of the workload group")
+	createGroupCmd.PersistentFlags().StringVarP(&namespace, "namespace", "n", "", "The namespace that the workload instances will belong to")
+	createGroupCmd.PersistentFlags().StringSliceVarP(&labels, "labels", "l", nil, "The labels to apply to the workload instances; e.g. -l env=prod,vers=2")
+	createGroupCmd.PersistentFlags().StringSliceVarP(&ports, "ports", "p", nil, "The incoming ports that the workload instances will expose")
+	createGroupCmd.PersistentFlags().StringVar(&network, "network", "default", "The name of the network for the workload instances")
+	createGroupCmd.PersistentFlags().StringVarP(&serviceAccount, "serviceAccount", "s", "default", "The service identity to associate with the workload instances")
+	return createGroupCmd
+}
+
+func generateWorkloadGroupYAML(u *unstructured.Unstructured, spec *networkingv1alpha3.WorkloadGroup) ([]byte, error) {
+	iSpec, err := unstructureIstioType(spec)
+	if err != nil {
+		return nil, err
+	}
+	u.Object["spec"] = iSpec
+
+	wgYAML, err := yaml.Marshal(u.Object)
+	if err != nil {
+		return nil, err
+	}
+	return wgYAML, nil
+}
+
+// Cluster inference from the current kubectl context only works for GKE
+func generateConfigCommand() *cobra.Command {
+	generateConfigCmd := &cobra.Command{
+		Use:   "generate-config",
+		Short: "Generates and packs all the required configuration files for deployment",
+		Long: `Takes in WorkloadGroup artifact, then generates and packs all the required configuration files for deployment. 
+This includes a MeshConfig resource, the cluster.env file, and necessary certificates and security tokens. 
+Tries to automatically infer the target cluster, and prompts for flags if the cluster cannot be inferred`,
+		Example: "generate-config -f workloadgroup.yaml -o config",
+		Args: func(cmd *cobra.Command, args []string) error {
+			if filename == "" {
+				return fmt.Errorf("expecting a WorkloadGroup artifact file")
+			}
+			if outputName == "" {
+				return fmt.Errorf("expecting an output filename")
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			wg := &clientv1alpha3.WorkloadGroup{}
+			if err := readWorkloadGroup(filename, wg); err != nil {
+				return err
+			}
+			// consider passed in flags before attempting to infer config
+			cluster, err := gkeCluster(gkeProject, clusterLocation, clusterName)
```

Comment:
highly suspicious to have a GKE specific command at the top level command. If we must have platform specific code, it should be abstracted away so the mainline code doesn't become `if gke {...} if eks {...} if rancher { ... }`

Reply:
thanks @howardjohn. Specific platform-isms should be rare within the codebase, and if so, they should really use some sort of plugin mechanism, much like cloud-provider in K8s. Otherwise, platform proliferation takes place. 👍